url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveParser_SelectClauseParser.java

```diff
@@ -0,0 +1,7251 @@
+	public static final int TOK_TIME=857;
+	public static final int TOK_TIMESTAMP=858;
+	public static final int TOK_TINYINT=859;
+	public static final int TOK_TMP_FILE=860;
+	public static final int TOK_TRANSFORM=861;
+	public static final int TOK_TRUE=862;
+	public static final int TOK_TRUNCATETABLE=863;
+	public static final int TOK_UNION=864;
+	public static final int TOK_UNIONTYPE=865;
+	public static final int TOK_UNIQUEJOIN=866;
+	public static final int TOK_UNLOCKTABLE=867;
+	public static final int TOK_USER=868;
+	public static final int TOK_USERSCRIPTCOLNAMES=869;
+	public static final int TOK_USERSCRIPTCOLSCHEMA=870;
+	public static final int TOK_VARCHAR=871;
+	public static final int TOK_VIEWPARTCOLS=872;
+	public static final int TOK_WHERE=873;
+	public static final int TOK_WINDOWDEF=874;
+	public static final int TOK_WINDOWRANGE=875;
+	public static final int TOK_WINDOWSPEC=876;
+	public static final int TOK_WINDOWVALUES=877;
+
+	// delegates
+	public Parser[] getDelegates() {
+		return new Parser[] {};
+	}
+
+	// delegators
+	public HiveParser gHiveParser;
+	public HiveParser gParent;
+
+
+	public HiveParser_SelectClauseParser(TokenStream input, HiveParser gHiveParser) {
+		this(input, new RecognizerSharedState(), gHiveParser);
+	}
+	public HiveParser_SelectClauseParser(TokenStream input, RecognizerSharedState state, HiveParser gHiveParser) {
+		super(input, state);
+		this.gHiveParser = gHiveParser;
+		gParent = gHiveParser;
+	}
+
+	protected TreeAdaptor adaptor = new CommonTreeAdaptor();
+
+	public void setTreeAdaptor(TreeAdaptor adaptor) {
+		this.adaptor = adaptor;
+	}
+	public TreeAdaptor getTreeAdaptor() {
+		return adaptor;
+	}
+	@Override public String[] getTokenNames() { return HiveParser.tokenNames; }
+	@Override public String getGrammarFileName() { return "SelectClauseParser.g"; }
+
+
+	  @Override
+	  public Object recoverFromMismatchedSet(IntStream input,
+	      RecognitionException re, BitSet follow) throws RecognitionException {
+	    throw re;
+	  }
+	  @Override
+	  public void displayRecognitionError(String[] tokenNames,
+	      RecognitionException e) {
+	    gParent.errors.add(new ParseError(gParent, e, tokenNames));
+	  }
+
+
+	public static class selectClause_return extends ParserRuleReturnScope {
+		CommonTree tree;
+		@Override
+		public CommonTree getTree() { return tree; }
+	};
+
+
+	// $ANTLR start "selectClause"
+	// SelectClauseParser.g:52:1: selectClause : ( KW_SELECT ( hintClause )? ( ( ( KW_ALL |dist= KW_DISTINCT )? selectList ) | (transform= KW_TRANSFORM selectTrfmClause ) ) -> {$transform == null && $dist == null}? ^( TOK_SELECT ( hintClause )? selectList ) -> {$transform == null && $dist != null}? ^( TOK_SELECTDI ( hintClause )? selectList ) -> ^( TOK_SELECT ( hintClause )? ^( TOK_SELEXPR selectTrfmClause ) ) | trfmClause -> ^( TOK_SELECT ^( TOK_SELEXPR trfmClause ) ) );
+	public final HiveParser_SelectClauseParser.selectClause_return selectClause() throws RecognitionException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 100. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.